tests/cases/string/like.test

```diff
@@ -0,0 +1,12 @@
+### SUBSTRAIT_SCALAR_TEST: v1.0
+### SUBSTRAIT_INCLUDE: '/extensions/functions_string.yaml'
+
+# basic: Basic examples without any special cases
+like('abcdefg'::str, 'abcdefg'::str) = true::bool
+like('abcdefg'::str, 'abc'::str) = false::bool
+
```

Comment:
future: add a test case with uppercase to verify that wildcards are case sensitive